packages/zowe-explorer/src/SpoolProvider.ts

```diff
@@ -145,10 +146,8 @@ export async function getSpoolFiles(node: IZoweJobTreeNode): Promise<zowe.IJobFi
  * @returns true if the selected node matches the spool file, false otherwise
  */
 export function matchSpool(spool: zowe.IJobFile, node: IZoweJobTreeNode): boolean {
-    return (
-        `${spool.stepname}:${spool.ddname} - ${spool["record-count"]}` === node.label.toString() ||
-        `${spool.stepname}:${spool.ddname} - ${spool.procstep}` === node.label.toString()
-    );
+    const nodeSpool = (node as ZoweSpoolNode).spool;
```

Comment:
If we have to cast `node` to `ZoweSpoolNode` to access the spool property, wouldn't this imply that `matchSpool` should be updated to only accept nodes that extend `ZoweSpoolNode`?

Reply:
Yes, although this would require changes across multiple methods.  We could change `nodes: IZoweJobTreeNode[]` to be `nodes: ZoweSpoolNode[]` on the downloadSingleSpool method - in theory this should be safe since the Download Spool action is only available for spool nodes 😋